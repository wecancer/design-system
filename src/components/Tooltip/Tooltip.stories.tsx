import React from 'react'
import {Story} from '@storybook/react'
import styled from 'styled-components'

import Tooltip, {Props} from '.'

const Wrap = styled.div`
  width: 100vw;
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
`

const Box = styled.div`
  width: 7.5rem;
  height: 7.5rem;
  color: #fff;
  font-weight: bold;
  background-color: red;
  transition: all ease 250ms;
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 0.375rem;
  cursor: pointer;

  &:hover {
    box-shadow: 0 0 1.25rem 0.25rem rgba(0, 0, 0, 0.3);
  }
`

const Template: Story<Props> = (args) => (
  <Wrap>
    <Tooltip {...args}>
      <Box>Hover me!</Box>
    </Tooltip>
  </Wrap>
)

export const Default = Template.bind({})

Default.args = {
  width: 240,
  children: 'hover me!',
  position: 'top',
  tooltip: 'Tooltip content',
}

const positions = ['top', 'bottom', 'left', 'right']

export default {
  title: 'Components/Tooltip',
  component: Tooltip,
  argTypes: {
    children: {
      control: {
        type: 'text',
      },
    },
    tooltip: {
      control: {
        type: 'text',
      },
    },
    position: {
      control: {
        type: 'select',
        options: positions,
      },
    },
  },
}
