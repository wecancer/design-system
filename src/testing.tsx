import React from 'react'
import {render, RenderResult} from '@testing-library/react'
import WecancerProvider from './WecancerProvider'

type Props = {
  children: React.ReactNode
}

const AllTheProviders = ({children}: Props) => {
  return <WecancerProvider>{children}</WecancerProvider>
}

// eslint-disable-next-line
const appRender = (ui: React.ReactElement, options?: any): RenderResult =>
  render(ui, {wrapper: AllTheProviders, ...options})

// re-export everything
export * from '@testing-library/react'

// override render method
export {appRender as render}
