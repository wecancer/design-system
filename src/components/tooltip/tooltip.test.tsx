import Tooltip from '.'
import {render, fireEvent} from '../../testing'

describe('<Tooltip />', () => {
  it('should render snapshot correctly', () => {
    const {container} = render(
      <>
        <Tooltip>
          <button type="button">hover me</button>
        </Tooltip>
        <Tooltip tooltip="testing...">
          <button type="button">hover me</button>
        </Tooltip>
        <Tooltip tooltip="top tooltip text" position="top">
          <button type="button">tooltip top</button>
        </Tooltip>
        <Tooltip tooltip="bottom tooltip text" position="bottom">
          <button type="button">tooltip bottom</button>
        </Tooltip>
        <Tooltip tooltip="left tooltip text" position="left">
          <button type="button">tooltip left</button>
        </Tooltip>
        <Tooltip tooltip="right tooltip text" position="right">
          <button type="button">tooltip right</button>
        </Tooltip>
      </>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should shows the tooltip when the mouse hover on element', () => {
    const {queryByTestId, container} = render(
      <Tooltip tooltip="tooltip testing...">
        <button type="button" data-testid="hover-me">
          button tooltip
        </button>
      </Tooltip>,
    )

    const hoverElement = queryByTestId('hover-me')
    fireEvent.mouseOver(hoverElement as HTMLElement)

    const tooltipContainer = container.querySelector('[role="tooltip"]')
    expect(tooltipContainer).toBeInstanceOf(HTMLElement)
  })
})
