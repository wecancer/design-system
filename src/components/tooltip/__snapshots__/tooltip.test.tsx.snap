// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> should render snapshot correctly 1`] = `
<div
  class="sc-bczRLJ cyTKsW"
>
  <button
    type="button"
  >
    hover me
  </button>
</div>
`;
